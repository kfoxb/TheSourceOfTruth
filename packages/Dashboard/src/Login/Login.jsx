import React from 'react';
import googleSignin from './google_signin.png';

export default function Login() {
  return (
    <div style={{
      height: '100vh', display: 'flex', alignItems: 'center', justifyContent: 'center',
    }}
    >
      <img alt="Sign in with Google" src={googleSignin} style={{ maxWidth: '50%' }} />
    </div>
  );
}
