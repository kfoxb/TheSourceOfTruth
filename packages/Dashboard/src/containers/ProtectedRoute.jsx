import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Route, Redirect } from 'react-router-dom';

class ProtectedRoute extends Component {
  protectRoute = () => {
    // eslint-disable-next-line no-unused-vars
    const { component = Component, isAuthenticated, ...props } = this.props;
    return (
      isAuthenticated === true
        ? <Component {...props} />
        : <Redirect to="/login" />
    );
  };

  render() {
    return (
      <Route
        render={this.protectRoute}
      />
    );
  }
}

ProtectedRoute.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  isAuthenticated: state.appState.isAuthenticated,
});

export default connect(mapStateToProps)(ProtectedRoute);
