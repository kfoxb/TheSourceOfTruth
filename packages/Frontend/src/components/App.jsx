import React, { Fragment } from 'react';
import { Route } from 'react-router-dom';
import { withRouter } from 'react-router';
import PropTypes from 'prop-types';
import { SidebarLeftOverlay, SidebarTopOverlay } from './Navigation';
import Home from './Home';
import Journal from './Journal';
import Library from './Library';
import SignInContainer from '../containers/SignInContainer';
import SignUp from './SignUp';

function App({
  isAuthenticated, logout, toggleVisibility, visible,
}) {
  return (
    <Fragment>
      <SidebarTopOverlay toggleMenu={toggleVisibility} />
      <SidebarLeftOverlay
        isAuthenticated={isAuthenticated}
        logout={logout}
        sideBarVisibility={visible}
        toggleMenu={toggleVisibility}
      />
      <div style={{ height: '40px' }} />
      <Route exact path="/" component={Home} />
      <Route path="/journal" component={Journal} />
      <Route path="/library" component={Library} />
      <Route path="/signin" component={SignInContainer} />
      <Route path="/signup" component={SignUp} />
    </Fragment>
  );
}

App.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  logout: PropTypes.func.isRequired,
  toggleVisibility: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default withRouter(App);
