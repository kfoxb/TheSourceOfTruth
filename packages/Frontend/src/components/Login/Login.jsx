import React from 'react';
import { Button } from 'semantic-ui-react';
import googleSignin from './google_signin.png';

export default function Login() {
  return (
    <div style={{
      height: '100vh', display: 'flex', alignItems: 'center', justifyContent: 'center',
    }}
    >
      <img alt="Sign in with Google" src={googleSignin} style={{ maxWidth: '50%' }} />
      <Button color="green">Sign in</Button>
    </div>
  );
}
