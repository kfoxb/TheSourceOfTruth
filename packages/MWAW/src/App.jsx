import React from 'react';
import styled from 'styled-components';

const Button = styled.button`
  display: inline-block;
  color: #FFFBFA;
  background: #666;
  border: none;
  padding: 0.5rem;
  width: 10rem;
  margin: 0.5rem;
  text-align: center;
`
const Menu = styled.div`

`
const Nav = styled.div`
  background: #666;
  opacity: 0.7;
  width: 100%;
  padding: 0.5em;
  margin: 0em;
  height: 45px;
  clear: right;
  z-index: 99;
`
const Search = styled.input`
  display: inline;
  color: #666;
  background: #FFFBFA;
  opacity: 0.8;
  border: 1px solid #666;
  padding: 0.5rem;
  width: 8rem;
  margin: 0.5rem;
  text-align: left;
  position: fixed;
  float: right;
  overflow: none;
  z-index: 100;
`
export default function App() {
  return (
    <Nav>
      <div>
        <Button>Login</Button>
        <Search
        placeholder=" Search"
        innerRef={x => { this.input = x }}
        onMouseEnter={() => this.input.focus()}
      />
      </div>
    </Nav>
  );
}
