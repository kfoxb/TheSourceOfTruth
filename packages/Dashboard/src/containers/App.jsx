import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Login from '../components/Login';
import Nav from '../components/Nav';

export function App({ isAuthenticated }) {
  if (!isAuthenticated) {
    return (<Login />);
  }
  return (
    <Nav />
  );
}

App.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  isAuthenticated: state.appState.isAuthenticated,
});

export default connect(mapStateToProps)(App);
