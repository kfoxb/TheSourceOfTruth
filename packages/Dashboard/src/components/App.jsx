import React, { Fragment } from 'react';
import { Route } from 'react-router-dom';
import ProtectedRoute from '../containers/ProtectedRoute';
import Login from './Login';
import Nav from './Nav';

export default function App() {
  return (
    <Fragment>
      <Route path="/login" component={Login} />
      <ProtectedRoute exact path="/" component={Nav} />
    </Fragment>
  );
}
