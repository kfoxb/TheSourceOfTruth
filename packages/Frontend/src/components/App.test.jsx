/* eslint-env mocha */
/* eslint-disable no-unused-expressions */
import React from 'react';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import { Route } from 'react-router-dom';
import App from './App';
import Login from './Login';
import Nav from './Nav';
import ProtectedRoute from '../containers/ProtectedRoute';

describe('App', () => {
  it('should a render', () => {
    shallow(<App />);
  });

  it('should render a Route with a login path that renders Login', () => {
    const wrapper = shallow(<App />);
    const loginRoute = wrapper.find(Route);
    expect(loginRoute).to.exist;
    expect(loginRoute.props().path).to.equal('/login');
    expect(loginRoute.props().component).to.equal(Login);
  });

  it('should render a ProtectedRoute with exact path / that renders Nav', () => {
    const wrapper = shallow(<App />);
    const rootRoute = wrapper.find(ProtectedRoute);
    expect(rootRoute).to.exist;
    expect(rootRoute.props().path).to.equal('/');
    expect(rootRoute.props().exact).to.equal(true);
    expect(rootRoute.props().component).to.equal(Nav);
  });
});
