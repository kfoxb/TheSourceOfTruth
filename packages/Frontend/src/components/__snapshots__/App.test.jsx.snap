// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<React.Fragment>
  <SidebarTopOverlay
    toggleMenu={[Function]}
  />
  <SidebarLeftOverlay
    sideBarVisibility={true}
    toggleMenu={[Function]}
  />
  <div
    style={
      Object {
        "height": "40px",
      }
    }
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/journals"
    />
    <Route
      path="/journals/edit/:id"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/journals/view/:id"
    />
    <Route
      component={[Function]}
      path="/library"
    />
    <Route
      component={[Function]}
      path="/sign(up|in)"
    />
  </Switch>
</React.Fragment>
`;
